### DISCUSSION TITLE
Best way to temporarily disable/pause dnscrypt-proxy Windows service?

### DISCUSSION THREAD

**Discussion by e-t-l** [DNSCrypt/dnscrypt-proxy#1727] [Q&A] [answered]
According to the [documentation](https://github.com/dyne/dnscrypt-proxy/blob/master/README-WINDOWS.markdown) for a fork by @dyne,
> If you changed the DNS settings for a given network interface to 127.0.0.1, you can revert these settings to what they used to be (or to any non-DNSCrypt resolver) anytime in order to stop using DNSCrypt. Changing the address back to 127.0.0.1 makes the network interface use the DNSCrypt proxy again.

Assuming this is the same for the default dnscrypt-proxy service and not something invented by @dyne, do I understand correctly that I could pause dnscrypt-proxy, e.g. via Powershell, with:
```
Set-DnsClientServerAddress -InterfaceIndex 9 -ResetServerAddresses
#stuff I want to do while paused here
Set-DnsClientServerAddress -InterfaceIndex 9 -ServerAddresses ("127.0.0.1")
```
(note: "9" is the interfaceindex for my use case. Your interface index can be found with `get-netadapter`)

If my assumption is NOT correct, then what is the best way to temporarily disable dnscrypt-proxy? (My backup plan is `Stop-Service -Name "dnscrypt-proxy"` / `Start-Service -Name "dnscrypt-proxy"`, but that seems a little brute-force.)

Thanks!

**Comment by lifenjoiner** [comment, marked as answer]
That's right.
It's a general model for a local DNS server: listening on a local port (53), and setting the system setting to use it.
And there seems no watch dog of the service on "setting the system setting to use it", it's manual.
So, just changing the system DNS settings will achieve your goal.

**Reply by e-t-l** [reply]
Thanks! And I just noticed the quote I gave above said "IF you changed the DNS settings for a given network interface to 127.0.0.1", implying that this is an example IP address, not necessarily the correct one. Is there any way to easily figure out what IP address my DNScrypt instance uses? (I run [Simple DNScrypt](https://github.com/bitbeans/SimpleDnsCrypt) because I'm not that knowledgeable about this stuff and appreciate the GUI, but unfortunately their documentation is a little sparse).

**Comment by lifenjoiner** [comment]
A local server always listens to [localhost](https://en.wikipedia.org/wiki/Localhost). `127.0.0.0/8` addresses are all equal. `127.0.0.1` is the most used.
You can just switch it between `127.0.0.1` and your temporary one.

**Reply by e-t-l** [reply]
Thank you! And I just confirmed that Simple DNScrypt [does indeed use 127.0.0.1](https://github.com/bitbeans/SimpleDnsCrypt/issues/304#issuecomment-383364278) as default.
